components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt

```diff
@@ -79,32 +79,36 @@ class AppLinksUseCases(
     ) {
         operator fun invoke(url: String): AppLinkRedirect {
             val intents = createBrowsableIntents(url)
-            val appIntent = if (includeHttpAppLinks) {
-                intents.firstOrNull { getNonBrowserActivities(it).isNotEmpty() }
+            val (appIntent, resolveInfos) = if (includeHttpAppLinks) {
+                intents.asSequence()
+                    .map { it to getNonBrowserActivities(it) }
+                    .firstOrNull { it.second.isNotEmpty() }
```

Comment:
This predicate should always be true since it doesn't seem possible to call `PackageManager#queryIntentActivities` and get back a list of null results.

Reply:
Done!

Reply:
Apparently this fails some tests/implementation, so I'm going to leave this the way it is.